File: java/parsers/src/main/java/jp/co/teruuu/parser/json/type/JArray.java
package jp.co.teruuu.parser.json.type;


import java.util.ArrayList;
import java.util.List;

public class JArray implements JValue {
  public List<JValue> value;

  public JArray() {
    this.value = new ArrayList<JValue>();
  }
  public JArray(List<JValue> value) {
    this.value = value;
  }
  public void add(JValue a) {
    this.value.add(a);
  }

  public boolean equals(Object other) {
    if(!(other instanceof JArray && this.value.size() == ((JArray) other).value.size())){
      return false;
    }
    for(int i = 0; i < this.value.size(); i++ ){
      if(!this.value.get(i).equals(((JArray)other).value.get(i))) {
        return false;
      }
    }
    return true;
  }
}

File: scala/parsers/src/main/scala/jp/co/teruuu/parser/json/type/JArray.scala
package jp.co.teruuu.parser.json.`type`

case class JArray(value: List[JValue]) extends JValue {
  def getVal = value

  override def equals(obj: Any): Boolean = {
    obj match {
      case obj: JArray => value.length == obj.getVal.length &&
        value.zipWithIndex.find{case (v, index) => !obj.getVal(index).equals(v)}.isEmpty
      case _ => false
    }
  }

  def toSpray(): spray.json.JsValue = {
    spray.json.JsArray(value.map(_.toSpray()).toVector)
  }

}
